cmd/nirvana/project/init.go

```diff
@@ -744,61 +798,101 @@ OUTPUT_DIR := ./bin
 # Build direcotory.
 BUILD_DIR := ./build
 
-# Git commit sha.
-COMMIT := $(strip $(shell git rev-parse --short HEAD 2>/dev/null))
-COMMIT := $(COMMIT)$(shell git diff-files --quiet || echo '-dirty')
-COMMIT := $(if $(COMMIT),$(COMMIT),"Unknown")
-{{ if .GoMod }}
-# This will force go to use the vendor files instead of using the $GOPATH/pkg/mod. (vendor mode)
-# more info: https://github.com/golang/go/wiki/Modules#how-do-i-use-vendoring-with-modules-is-vendoring-going-away
-export GOFLAGS := -mod=vendor
-{{ end }}
+# Current version of the project.
+VERSION ?= $(shell git describe --tags --always --dirty)
+
+# Available cpus for compiling, please refer to https://github.com/caicloud/engineering/issues/8186#issuecomment-518656946 for more information.
+CPUS ?= $(shell /bin/bash hack/read_cpus_available.sh)
+
+# Track code version with Docker Label.
+DOCKER_LABELS ?= git-describe="$(shell date -u +v%Y%m%d)-$(shell git describe --tags --always --dirty)"
+
+# Default golang flags used in build and test
+# -mod=vendor: force go to use the vendor files instead of using the` + " `$GOPATH/pkg/mod` " + `
+# -p: the number of programs that can be run in parallel
+# -race: enable data race detection
+# -count: run each test and benchmark 1 times. Set this flag to disable test cache
+export GOFLAGS ?= -mod=vendor -p=$(CPUS) -race -count=1
+
+# Golang standard bin directory.
+GOPATH ?= $(shell go env GOPATH)
+BIN_DIR := $(GOPATH)/bin
+GOLANGCI_LINT := $(BIN_DIR)/golangci-lint
```

Comment:
Keep the same order of these variables with template, it would be better find the changes through diff.

Reply:
Updated, PTAL